app/src/main/java/com/postpc/tenq/services/RecorderService.java

```diff
@@ -0,0 +1,138 @@
+package com.postpc.tenq.services;
+
+import android.content.Context;
+import android.media.AudioManager;
+import android.media.MediaRecorder;
+import android.util.Log;
+import android.os.Handler;
+
+import static java.lang.Math.abs;
+import static java.lang.Math.exp;
+
+public class RecorderService {
+
+    private static double mEMA = 0.0;
+    static final private double EMA_FILTER = 0.6;
+
+    private MediaRecorder mRecorder;
+    private final Context applicationContext;
+    private static Thread runner;
+    private boolean isRecorderOn = true; // TODO: change to false when not debugging
+    private double prevAmplitude;
+    private AudioManager audioManager;
+
+    public RecorderService(Context applicationContext) {
+
+        this.applicationContext = applicationContext;
```

Comment:
Why are you saving a reference to the app context? I see it's only used in the next line

Reply:
I'm using it in the next line to create an AudioManager object to control the volume.

Reply:
But then never using the app context again? So you don't need to set `this.applicationContext = applicationContext`